### PR TITLE
Downgrade required Python to 3.10

### PR DESCRIPTION
**Pull request by aarmea-hyperfine**
This doesn't seem to use any 3.11 features and works just fine on 3.10.

This is probably as low as you can go without any code changes -- from what I could tell in my 30 second audit, Python 3.9 support would only require replacing the `|`s in the type hints with `typing.Union`.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -14,7 +14,7 @@ include = ["LICENSE", "README.md"]
 crosshash-py = 'crosshash:main'
 
 [tool.poetry.dependencies]
-python = "^3.11"
+python = "^3.10"
```
Comment (janbrasna) [inline pyproject.toml:17]: While yeah, it is compatible with 3.10 fine, just tested it, have you actually tried running `make install` on a system where no 3.11 binary is available? Esp. in CI this:

https://github.com/httpie/crosshash/blob/a355b74ed452d26bc49d9f4d7d4f3a1626f2f2de/Makefile#L14-L17

would bail with: 

``` sh
poetry env use python3.11
> Could not find the python executable python3.11
> make: *** [Makefile:16: install-py] Error 1
> Error: Process completed with exit code 2.
```

if run with other versions.

TL;DR to make this complete, not break for 3.10 folks, and support e.g. 3.12 along the way, changes to `Makefile` might be necessary. (My guess is that just omitting the `poetry env` and leaving that to magic defaults should work. It will create an env based on what python version was used to install it, and if that one is not compatible with project requirements poetry tries activating a new one better suited automagically; if it fails here it prompts the user what to do at that point. Better than exit code for unavailable binary;)…)

Are you comfortable fiddling with this, or need a hand?

Comment (janbrasna) [reply pyproject.toml:17]: OK so I went ahead and [left out](https://github.com/janbrasna/crosshash/commit/129238620cce43f89635a7c6f7a71f9092364ce3) the explicit version poetry env and it picks up the respective versions correctly:
1. [`Creating virtualenv crosshash-qqERbTL5-py3.10`](https://github.com/janbrasna/crosshash/actions/runs/10431134610/job/28890474111#step:5:142)
2. [`platform linux -- Python 3.10.14, pytest-7.4.0, pluggy-1.3.0`](https://github.com/janbrasna/crosshash/actions/runs/10431134610/job/28890474111#step:6:12)

> Screen Shot 2024-08-17 at 13 34 51

Would you remember @jkbrzt why is the env constraint manually set in Makefile like this? Is that any help for local work perhaps? Generally poetry finds a way without telling it at all, lower constraint is well defined in pyproject dependency — e.g. [here for incompatible env (3.9)](https://github.com/janbrasna/crosshash/actions/runs/10431164106/job/28890529255#step:6:12) than specified dep it resolves that looking for next matching version when available (and when not available reasonably prompts the users i.e. no hard failures, pretty nice UX/DX) — and no upper constraint is currently necessary as I've run it on [3.13-rc.1 just fine](https://github.com/janbrasna/crosshash/actions/runs/10431314237/job/28890835365#step:6:12) FWIW… So as far I can tell, the env could/should be left to poetry implicit magic, allowing using `make install(-py)` even with different versions.

Screen Shot 2024-08-17 at 13 29 39

Screen Shot 2024-08-17 at 13 30 28
